api/src/api/cookie-category/services/cookie-category.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+/**
+ * cookie-category service
+ */
+
+const { createCoreService } = require('@strapi/strapi').factories;
+
+module.exports = createCoreService(
+  'api::cookie-category.cookie-category',
+  ({ strapi }) => ({
+    async find(params) {
+      const { pagination } = await super.find(params);
+      const results = await strapi.entityService.findMany(
+        'api::cookie-category.cookie-category',
```

Comment:
is this just setting the default population? If so can we move this to a route middleware instead? not sure it really makes sense keep this as a controller. This is also fairly inefficient since you are duplicating the total DB calls here for cookies. When you trigger the `super.find()` you are still finding the cookies and then you do the exact same request just below it. It will take twice as long with this controller. Instead in a route middleware we could just set the `ctx.query.populate = ['cookies']` and assign it to the route. Cleaner and easier implementation that doesn't impact performance.

Reply:
I will no have the time to dig this tbh, I rather remove the plugin if you consider it not having good practices inside of it.